handler/processors.go

```diff
@@ -422,6 +422,57 @@ func processCheckRunEvent(log *logrus.Entry, token string, event *github.CheckRu
 	return targetEntities, eventDetails, nil
 }
 
+func processCheckSuiteEvent(log *logrus.Entry, token string, event *github.CheckSuiteEvent) ([]*TargetEntity, *EventDetails, error) {
+	log.Info("processing check_suite event")
+
+	targetEntities := []*TargetEntity{}
+
+	eventDetails := &EventDetails{
+		EventName:   "check_suite",
+		EventAction: event.GetAction(),
+		Payload:     event,
+	}
+
+	//  if the head is from a forked repository the pull_requests array will be empty on check suite events
+	if len(event.CheckSuite.PullRequests) == 0 {
+		prs, err := getPullRequests(token, event.GetRepo().GetFullName())
+		if err != nil {
+			return nil, nil, err
+		}
+
+		log.Infof("fetched %d pull requests", len(prs))
+
+		for _, pr := range prs {
+			if pr.GetHead().GetSHA() == event.CheckSuite.GetHeadSHA() {
+				log.Infof("found pull request %v", pr.GetNumber())
+				return []*TargetEntity{
+					{
+						Kind:   PullRequest,
+						Number: pr.GetNumber(),
+						Owner:  event.GetRepo().GetOwner().GetLogin(),
+						Repo:   event.GetRepo().GetName(),
+					},
+				}, eventDetails, nil
+			}
+		}
+
+		log.Infof("no pr found with the head sha %v", event.CheckSuite.GetHeadSHA())
+
+		return []*TargetEntity{}, eventDetails, nil
+	}
+
+	for _, pr := range event.CheckSuite.PullRequests {
+		targetEntities = append(targetEntities, &TargetEntity{
+			Kind:   PullRequest,
+			Owner:  event.GetRepo().GetOwner().GetLogin(),
+			Repo:   event.GetRepo().GetName(),
+			Number: pr.GetNumber(),
+		})
+	}
```

Comment:
Let's include this code inside an `else` block.

```suggestion
	} else {
		for _, pr := range event.CheckSuite.PullRequests {
			targetEntities = append(targetEntities, &TargetEntity{
				Kind:   PullRequest,
				Owner:  event.GetRepo().GetOwner().GetLogin(),
				Repo:   event.GetRepo().GetName(),
				Number: pr.GetNumber(),
			})
		}
	}
```